src/error.rs

```diff
@@ -69,6 +71,43 @@ impl Error {
             _ => false,
         }
     }
+
+    /// Whether a read operation should be retried if this error occurs
+    pub(crate) fn is_read_retryable(&self) -> bool {
+        if self.is_network_error() {
+            return true;
+        }
+        let error_kind: &ErrorKind = &self.kind;
+        match &self.kind.code_and_message() {
+            Some((code, message)) => {
+                if RETRYABLE_READ_CODES.contains(&code) {
+                    return true;
+                }
+                if message.contains("not master") || message.contains("node is recovering") {
```

Comment:
we actually have methods on `ErrorKind` for this already: `is_not_master` and `is_recovering`

Reply:
The [spec indicates](https://github.com/mongodb/specifications/blob/5979b36b16e37427d9d49c0825944a02ef665a10/source/retryable-reads/retryable-reads.rst#id7) that any error that contains either of these specific substrings should be considered a retryable error -- super nitpicky, but would it be an issue that `is_recovering` also allows for the String "not master or secondary"?

Reply:
Nope that should be fine, since if "not master" itself is a substring of "not master or secondary", the union of our `is_recovering` and `is_not_master` methods is equivalent to the two substring checks you have here.